packages/react-portal/src/components/PortalProvider/usePortalProvider.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { PortalProviderProps, portalProviderShorthandProps, PortalProviderState } from './PortalProvider.types';
+import { usePortalMountNode } from '../../usePortalMountNode';
+
+const mergeProps = makeMergeProps<PortalProviderState>({ deepMerge: portalProviderShorthandProps });
+
+/**
+ * Create the state required to render PortalProvider.
+ *
+ * The returned state can be modified with hooks such as usePortalProviderStyles,
+ * before being passed to renderPortalProvider.
+ *
+ * @param props - props from this instance of PortalProvider
+ * @param ref - reference to root HTMLElement of PortalProvider
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ */
+export const usePortalProvider = (
+  props: PortalProviderProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: PortalProviderProps,
+): PortalProviderState => {
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef(null)),
```

Comment:
This looks really strange as in `renderPortal()` nothing renders `slots.root`. We should either cleanup `*Props` & `*State` interfaces or render `slots.root`.

Reply:
Added a TODO to discuss the root slot